packages/astro/src/manifest/create.ts

```diff
@@ -0,0 +1,283 @@
+
+
+			if (name[0] === '_') {
+				return;
+			}
+      if (basename[0] === '.' && basename !== '.well-known') {
+        return;
+      }
+      if (!isDir && !/^(\.[a-z0-9]+)+$/i.test(ext)) {
+        return; // filter out tmp files etc
+      }
+      const segment = isDir ? basename : name;
+      if (/^$/.test(segment)) {
+        throw new Error(`Invalid route ${file} — Collections API was deprecated and replaced with getStaticPaths.`);
+      }
+      if (/\]\[/.test(segment)) {
+        throw new Error(`Invalid route ${file} — parameters must be separated`);
+      }
+      if (countOccurrences('[', segment) !== countOccurrences(']', segment)) {
+        throw new Error(`Invalid route ${file} — brackets are unbalanced`);
+      }
+      if (/.+\[\.\.\.[^\]]+\]/.test(segment) || /\[\.\.\.[^\]]+\].+/.test(segment)) {
+        throw new Error(`Invalid route ${file} — rest parameter must be a standalone segment`);
+      }
+
+      const parts = getParts(segment, file);
+      const isIndex = isDir ? false : basename.startsWith('index.');
+      const routeSuffix = basename.slice(basename.indexOf('.'), -ext.length);
+
+      items.push({
+        basename,
+        ext,
+        parts,
+        file: slash(file),
+        isDir,
+        isIndex,
+        isPage: true,
+        routeSuffix,
+      });
+    });
+    items = items.sort(comparator);
+
+    items.forEach((item) => {
+      const segments = parentSegments.slice();
+
+      if (item.isIndex) {
+        if (item.routeSuffix) {
+          if (segments.length > 0) {
+            const lastSegment = segments[segments.length - 1].slice();
+            const lastPart = lastSegment[lastSegment.length - 1];
+
+            if (lastPart.dynamic) {
+              lastSegment.push({
+                dynamic: false,
+                spread: false,
+                content: item.routeSuffix,
+              });
+            } else {
+              lastSegment[lastSegment.length - 1] = {
+                dynamic: false,
+                spread: false,
+                content: `${lastPart.content}${item.routeSuffix}`,
+              };
+            }
+
+            segments[segments.length - 1] = lastSegment;
+          } else {
+            segments.push(item.parts);
+          }
+        }
+      } else {
+        segments.push(item.parts);
+      }
+
+      const params = parentParams.slice();
+      params.push(...item.parts.filter((p) => p.dynamic).map((p) => p.content));
+
+      if (item.isDir) {
+        walk(path.join(dir, item.basename), segments, params);
+      } else if (item.isPage) {
+        components.push(item.file);
+        const component = item.file;
+        const pattern = getPattern(segments, false);
+        const generate = getGenerator(segments, false);
+        const path = segments.every((segment) => segment.length === 1 && !segment[0].dynamic) ? `/${segments.map((segment) => segment[0].content).join('/')}` : null;
```

Comment:
I have an incredibly-small nit about `path` being reserved for the built-in. While TypeScript is smart enough to catch everything, and I don’t think it’ll ever cause a bug, IMO it’s just a minor readability annoyance to have `path` meaning different things in different contexts and having to be aware of where each was defined.

What do you think about `pathname` instead to refer to URL paths? Consistent with `new URL('…').pathname` ?

Reply:
+1, I usually do that myself!